curp/src/client/unary.rs

```diff
@@ -687,6 +621,124 @@ impl<C: Command> ClientApi for Unary<C> {
     }
 }
 
+#[async_trait]
+impl<C: Command> RepeatableClientApi for Unary<C> {
+    /// Generate a unique propose id during the retry process.
+    async fn gen_propose_id(&self) -> Result<ProposeId, Self::Error> {
+        let client_id = self.get_client_id().await?;
+        let seq_num = self.new_seq_num();
+        Ok(ProposeId(client_id, seq_num))
+    }
+
+    /// Send propose to the whole cluster, `use_fast_path` set to `false` to fallback into ordered
+    /// requests (event the requests are commutative).
+    async fn propose(
+        &self,
+        propose_id: ProposeId,
+        cmd: &Self::Cmd,
+        use_fast_path: bool,
+    ) -> Result<ProposeResponse<Self::Cmd>, Self::Error> {
+        tokio::pin! {
+            let fast_round = self.fast_round(propose_id, cmd);
+            let slow_round = self.slow_round(propose_id);
+        }
+
+        let res: ProposeResponse<C> = if use_fast_path {
+            match futures::future::select(fast_round, slow_round).await {
+                futures::future::Either::Left((fast_result, slow_round)) => match fast_result {
+                    Ok(er) => er.map(|e| (e, None)),
+                    Err(fast_err) => {
+                        if fast_err.should_abort_slow_round() {
+                            return Err(fast_err);
+                        }
+                        // fallback to slow round if fast round failed
+                        let sr = match slow_round.await {
+                            Ok(sr) => sr,
+                            Err(slow_err) => {
+                                return Err(std::cmp::max_by_key(fast_err, slow_err, |err| {
```

Comment:
Is it better to implement Ord for the errors?

Reply:
`ErrorPriority` did that